api/src/main/java/org/eclipse/microprofile/config/inject/ConfigProperties.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.config.inject;
+
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.enterprise.util.Nonbinding;
+import javax.inject.Qualifier;
+
+/**
+ * <p>
+ * Retrieve a number of related configuration properties with the specified prefix into a property class.
+ * <pre>
+ * &#064;ConfigProperties(prefix="server")
+ * public class MyServer {
+ *      public String host; //maps the property name server.host
+ *      public int port;    //maps to the property name server.port
+ *      private String context; //maps to the property name server.context
+ *      public &#064;ConfigProperty(name="old.location")
+ *      String location; //maps to the property name server.old.location
+ *      public String getContext() {
+ *          return context;
+ *      }
+ * }
+ * </pre>
+ *
+ * 
+ *
+ * @since 2.0
+ * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
+ * 
+ */
+
+@Target({METHOD, FIELD, PARAMETER, TYPE})
```

Comment:
This one should be `TYPE` only, right?

Reply:
Since `@ConfigProperties` can be used on field, method and parameters to specify the injection point, as well as classes, we need to have all of them.

Reply:
But is not mandatory, right? And it is only if you want to override the prefix?

Reply:
I don't quite get what you meant. If I remove the Field, Parameter, Method, you won't be able to do 
`@Inject @ConfigProperties` on a field, method or as a parameter. The following won't compile
`@Inject @ConfigProperties ServiceDetails sd;`

Reply:
I was trying to clarify if you do require to use `@ConfigProperties` to perform injection. The bean itself is already qualified with the annotation.

Reply:
ah. I see what you meant. Since you comment on this Target line, I interpret you ask for removing the other three. 



> I was trying to clarify if you do require to use `@ConfigProperties` to perform injection. The bean itself is already qualified with the annotation.

No, you don't. The spec has an example for it without the need to mention `@ConfigProperty` if you don't want to pass in any other prefix. You can do `@Inject ServiceDetails sd;`.

Reply:
Here is what spec states:

> 

    Directly inject the bean annotated with @ConfigProperties

> @Inject Details serverDetails;